pkg/types/preferencetypes/preference.go

```diff
@@ -222,6 +232,27 @@ func (p *Preference) IsValidValue(preferenceValue interface{}) error {
 				return errors.Newf(errors.TypeInvalidInput, errors.CodeInvalidInput, fmt.Sprintf("the preference value is not in the range specified, min: %v , max:%v", p.Range.Min, p.Range.Max))
 			}
 		}
+	case PreferenceValueTypeJSON:
```

Comment:
it should be something like preferenceValuteTypeAttributeKeys instead of generic json.